enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,313 @@
+      this means running OCP on the NICs, this is not yet cast in
+      stone. Some other approaches are discussed under "Alternatives"
+      and one of those could potentially replace the OCP approach.
+
+    - Although this enhancement technically only covers the
+      installing/maintaining part of the problem, there will be heavy
+      overlap with the "Smart NIC OVN Offload" enhancement (with the
+      decisions that are made here affecting how OVN and
+      ovn-kubernetes get deployed there).
+
+    - Although not part of the initial design, we assume some
+      customers may be interested taking advantage of the NIC/host
+      separation to run additional security and monitoring
+      functionality on the NICs in the future. Thus, we are interested
+      in designs with more isolation between the NIC and the host,
+      even though that is not necessarily interesting for the primary
+      network-offload use case.
+
+- Design an appropriate architecture for an environment containing a
+  mix of x86_64 hosts with "dumb" NICs, x86_64 hosts with Smart NICs,
+  and the ARM-based Smart NICs themselves. (As discussed below, such
+  an environment may involve one or more OCP clusters.)
+
+- Define how OCP installs and upgrades will work in such an
+  environment.
+
+[Smart NIC OVN Offload]: https://github.com/openshift/enhancements/pull/732
+
+### Non-Goals
+
+- Support for network plugins other than OVN Kubernetes. The
+  architecture designed here may not be generically applicable and is
+  not guaranteed to be usable by other network plugins even if they
+  independently implement their own BlueField, etc, support.
+
+### Supported Hardware
+
+We want to support Smart NICs that are vaguely "BlueField-2-ish". eg:
+
+  1. The NIC has a sufficiently-powerful ARM processor (or,
+     theoretically, a processor of some other architecture that RHCOS
+     supports), and sufficient RAM, SSD, etc, to run as an OCP node.
+
+       - Current BlueField-2 specs mention "Up to 8 Armv8 A72 cores"
+         and "8GB / 16GB / 32GB of on-board DDR4". It is possible we
+         would only support a subset of SKUs.
+
+       - The terms "Smart NIC" and "DPU" are also being used to refer
+         to NICs that just have programmable ASICs or FPGAs, rather
+         than a full computing environment. These are out of scope.
+
+  2. The NIC has an IPMI-addressable Baseboard Management Controller
+     that can be used to control and provision it independently of
+     its host.
+
+  3. The NIC implements SR-IOV in such a way that we can connect
+     pods on the host directly to an OVN bridge on the NIC.
+
+  4. All of the above can be done without running any proprietary
+     drivers or third-party software on either the host's or the NIC's
+     RHCOS installation.
+
+## Design Details
+
+### Clusters Will Contain a Mix of Smart and "Dumb" NICs
+
+OVN Kubernetes is not currently capable of mixing offloaded and
```

Comment:
@danwinship I think we should try to change this. It's not clear to me (though I haven't read deeply) why a pod would care that it was offloaded or not. So sure, pods could request being scheduled to an offloaded node, but by default I'd presume they'd land somewhere and if that happened to be a smartnic node, so be it.